WalletWasabi/BranchNBound/BranchAndBound.cs

```diff
@@ -0,0 +1,178 @@
+using NBitcoin;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.BranchNBound
+{
+	/// <seealso href="https://murch.one/wp-content/uploads/2016/11/erhardt2016coinselection.pdf">Section "5.3 Branch and Bound".</seealso>
+	public class BranchAndBound
+	{
+		private readonly Random _random = new();
+
+		public BranchAndBound(List<long> values)
+		{
+			Count = values.Count;
+			SortedValues = values.OrderByDescending(x => x).ToArray();
```

Comment:
The code assumes in many places that the `SortedValues` are positive numbers but that constrain is not enforced anywhere and this is where we should check it.

Reply:
Yeah, it should. Added.